src/sabre_swap/sabre_dag.rs

```diff
@@ -33,18 +33,18 @@ impl SabreDAG {
     pub fn new(
         num_qubits: usize,
         num_clbits: usize,
-        nodes: Vec<(usize, Vec<usize>, Vec<usize>)>,
+        nodes: Vec<(usize, Vec<usize>, HashSet<usize>)>,
         front_layer: PyReadonlyArray1<usize>,
     ) -> PyResult<Self> {
         let mut qubit_pos: Vec<usize> = vec![usize::MAX; num_qubits];
         let mut clbit_pos: Vec<usize> = vec![usize::MAX; num_clbits];
         let mut reverse_index_map: HashMap<usize, NodeIndex> = HashMap::with_capacity(nodes.len());
-        let mut dag: DiGraph<(usize, Vec<usize>, Vec<usize>), ()> =
+        let mut dag: DiGraph<(usize, Vec<usize>), ()> =
```

Comment:
If we don't need to track clbit order within the `SabreDAG` (which I think you're right - we don't need to), I don't think we need to track qubit order either - it's not important for the Sabre algorithm.  That said, it's not really worth bothering about, and if doing it becomes a memory/performance concern in the future we can revisit it.